public/static/app/perf-bb/SignalProcessingCtrl.js

```diff
@@ -234,6 +233,23 @@ mciModule.controller('SignalProcessingCtrl', function(
     );
   }
 
+  vm.onFilterChanged = function (api) {
+    Settings.perf.signalProcessing.persistentFiltering = _.reduce(api.grid.columns, function (m, d) {
+      if (d.visible) {
+        const term = d.filters[0].term;
+        if (term) {
+          m[d.field] = term;
+        }
+      }
+      return m;
+    }, {});
+
+    state.filtering = vm.changePointFilter();
+    vm.setInitialGridFiltering(vm.gridApi, state);
```

Comment:
Removing the if check around this is the only real change; everything else done was so that this can be tested. I couldn't really understand why that if check was being done in the original code -- do you know if there's some edge case I'm not seeing here?